tests/blackbox/testlib/dbus.py

```diff
@@ -0,0 +1,232 @@
+        """
+        pool_objects = [
+            obj_data[StratisDbus._POOL_IFACE]
+            for _, obj_data in StratisDbus._get_managed_objects().items()
+            if StratisDbus._POOL_IFACE in obj_data
+            and obj_data[StratisDbus._POOL_IFACE]["Name"].startswith(TEST_PREF)
+        ]
+
+        return [pool_obj["Name"] for pool_obj in pool_objects]
+
+    @staticmethod
+    def pool_destroy(pool_name):
+        """
+        Destroy a pool
+        :param name: Name of pool to destroy
+        :return: None
+        """
+        pool_objects = {
+            path: obj_data[StratisDbus._POOL_IFACE]
+            for path, obj_data in StratisDbus._get_managed_objects().items()
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+
+        pool_object_paths = [
+            path
+            for path, pool_obj in pool_objects.items()
+            if pool_obj["Name"] == pool_name and pool_obj["Name"].startswith(TEST_PREF)
+        ]
+        if pool_object_paths == []:
+            return None
+
+        iface = dbus.Interface(
+            StratisDbus._BUS.get_object(StratisDbus._BUS_NAME, StratisDbus._TOP_OBJECT),
+            StratisDbus._MNGR_IFACE,
+        )
+        iface.DestroyPool(pool_object_paths[0], timeout=StratisDbus._TIMEOUT)
+
+        return None
+
+    @staticmethod
+    def fs_list():
+        """
+        Query the file systems
+        :return: A dict,  Key being the fs name, the value being the pool name
+        """
+        objects = StratisDbus._get_managed_objects().items()
+
+        fs_objects = [
+            obj_data[StratisDbus._FS_IFACE]
+            for _, obj_data in objects
+            if StratisDbus._FS_IFACE in obj_data
+            and obj_data[StratisDbus._FS_IFACE]["Name"].startswith(TEST_PREF)
+        ]
+
+        pool_path_to_name = {
+            obj: obj_data[StratisDbus._POOL_IFACE]["Name"]
+            for obj, obj_data in objects
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+
+        result = {
+            fs_object["Name"]: pool_path_to_name[fs_object["Pool"]]
+            for fs_object in fs_objects
+        }
+
+        return result
+
+    @staticmethod
+    def fs_destroy(pool_name, fs_name):
+        """
+        Destroy a FS
+        :param pool_name:  Pool which contains the FS
+        :param fs_name: Name of FS to destroy
+        :return: None
+        """
+        objects = StratisDbus._get_managed_objects().items()
+
+        pool_objects = {
+            path: obj_data[StratisDbus._POOL_IFACE]
+            for path, obj_data in objects
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+        fs_objects = {
+            path: obj_data[StratisDbus._FS_IFACE]
+            for path, obj_data in objects
+            if StratisDbus._FS_IFACE in obj_data
+        }
+
+        pool_object_paths = [
+            path
+            for path, pool_obj in pool_objects.items()
+            if pool_obj["Name"] == pool_name
```

Comment:
Might as well match ```TEST_PREF``` here, for consistency.

Reply:
I've made some changes all around in the way that I do this in the most recent commit. I've started filtering in all cases on objects from `GetManagedObjects` before they're used anywhere. I think this is probably best to do right at the beginning so any code that gets added and needs to use those objects in testing doesn't need to worry about filtering independently. Resolved.